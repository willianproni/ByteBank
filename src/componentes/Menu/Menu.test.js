import { render, screen } from '@testing-library/react';
import Menu from '.';

describe('<Menu/>', () => {
  it('Deve renderizar um link para a página inicial', () => {
    render(<Menu />);
    expect(screen.getByText('Inicial')).toBeInTheDocument();
  });

  it('Renderizar todos os links', () => {
    render(<Menu />);
    expect(screen.getAllByRole('link')).toHaveLength(4);
  });

  it('Não deve renderizar o link para extrato', () => {
    render(<Menu />);
    expect(screen.queryByText('extrato')).not.toBeInTheDocument();
  });

  it('Renderizar os link e verificar se estão com className link', () => {
    render(<Menu />);
    const links = screen.getAllByRole('link');
    links.forEach((link) => {
      expect(link).toHaveClass('link');
    });
    expect(links).toMatchSnapshot();
  });

  //   it('Hover ao passar o mouse pelos titulos do menu', () => {
  //     render(<Menu />);
  //     fireEvent.mouseOver(screen.getByText('Inicial'));
  //     expect(screen.getByText('Inicial')).toHaveStyle({
  //       color: 'rgb(0, 0, 0)',
  //     });
  //   });
});
