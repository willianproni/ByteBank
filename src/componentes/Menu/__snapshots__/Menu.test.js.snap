// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu/> Renderizar os link e verificar se estão com className link 1`] = `
Array [
  <a
    class="link"
    href="/"
  >
    Inicial
  </a>,
  <a
    class="link"
    href="/"
  >
    Transferências
  </a>,
  <a
    class="link"
    href="/"
  >
    Investimentos
  </a>,
  <a
    class="link"
    href="/"
  >
    Outros serviços
  </a>,
]
`;
